### ISSUE TITLE
A little question about details of training and test

### ISSUE THREAD

**Issue by CHMimilanlan** [olivia23333/E3Gen#7]
Since I am new in this realm, and I am not familiar with the code of MMCV, I can't make fully understand of the training process. So I have a little question about this. 

At the beginning, I read carefully about the code diffusion_nerf.py as well as the function `val_uncond` , and I found that the variable `noise` is randomly sampled, and the diffusion process only take the `noise`, which generated by `torch.randn`, as input without any prior, and diffusion can process the UV latent code with the input of noise. With the decoder taking UV latent code as input, a real UV Map that refer to a specific man can be generated. 

So here is my question, why the diffusion process can generate a specific UV latent code without taking any prior. 
And my guess to this question is, the diffusion module is trained on specific man, so that the diffusion can process specific UV latent. If I want to generate a man out-of-dataset,  I can just fine-tune the diffusion modules. 
Is my guess correct or is there something I am analyzing wrong? I will be grateful for your timely response!

**Comment by olivia23333** [comment]
Our diffusion model works is similar to diffusion models used in 2D image generation. The main difference is that they map Gaussian noise to 2D images while ours map Gaussian noise to feature maps. You don't need to fine-tune diffusion modules and can generate various people with different Gaussian noises. I'm not sure why priors are needed for generations or what might that prior mean, maybe you can give some examples? If you want to know why noise can become reasonable output I recommend this repo and its corresponding paper: https://github.com/Stability-AI/stablediffusion?tab=readme-ov-file

**Comment by CHMimilanlan** [comment]
> Our diffusion model works is similar to diffusion models used in 2D image generation. The main difference is that they map Gaussian noise to 2D images while ours map Gaussian noise to feature maps. You don't need to fine-tune diffusion modules and can generate various people with different Gaussian noises. I'm not sure why priors are needed for generations or what might that prior mean, maybe you can give some examples? If you want to know why noise can become reasonable output I recommend this repo and its corresponding paper: https://github.com/Stability-AI/stablediffusion?tab=readme-ov-file

为了方便交流，我就直接用中文说了。
我理解diffusion的机制，以及在这里的作用是生成UV latent map，并且diffusion模型只能生成在训练数据集之内的人的UV latent。在推理的过程中（运行test.py --mode "test"），diffusion的输入是一个随机采样的噪声，输出是一个uv latent map。但是，经过我的反复实验，我发现在特定的iteration下，只能生成特定外观的人。 
![image](https://github.com/user-attachments/assets/40da9ede-f00c-40ea-b92c-ec1001bea4cd)
就比如，在第一个iteration中（如上图所示），经过diffusion的去噪过程，以及两个decoder，就一定会生成下面四位。
![image](https://github.com/user-attachments/assets/ce8f3133-ebd6-4781-a235-9aad35bf10ac)
不管经过多少次实验都是这四位。
那么我的问题是，既然输入的噪声是随机采样的，而且diffusion在去噪的过程中没有加入任何condition（就比如文生图中，text是作为diffusion的condition用于控制图片的生成），那么这里diffusion的过程没有condition约束就应该是一个uncondional image generation，生成的uv latent也不应该具备指代特定人的特征。但是，这些uv latent在经过decoder之后就一定会生成特定的人，所以这就是我疑惑的地方，为什么没有任何condition去噪得到的uv latent就能生成特定的人。

非常期待您的回复！

**Comment by olivia23333** [comment]
因为random seed是固定的，也就是说每到这个iter用的random noise会是一样的，加一个--seed 100或者其他数字设置不同的random seed之后，你就会得到不一样的4个人

**Comment by CHMimilanlan** [comment]
> 因为random seed是固定的，也就是说每到这个iter用的random noise会是一样的，加一个--seed 100或者其他数字设置不同的random seed之后，你就会得到不一样的4个人


好的，我理解了，非常感谢！
